Validate image passed to ukvm is in ELF format

myechuri opened the pull request:
Since ukvm expects its payload to be in ELF format, validate that the ELF header has expected magic characters. Without the validation, any image format (ex: qcow2, which is the default OSv image format) is loaded successfully, and ``vcpu_loop`` execution fails with ``KVM_EXIT_SHUTDOWN`` ([Issue 13](https://github.com/Solo5/solo5/issues/88)).

This PR attempts to validate ELF magic characters in the loaded code so that -
a) user is informed of mismatched image format
b) ukvm does not attempt to get into ``vcpu_loop`` with a corrupt guest
___________
Testing Done:

Invalid image without validation:
```
root@ubuntu:~/solo5/tests/test_hello# ./ukvm-bin /tmp/osv-images/ukvm.img 
Setting the CPUID.
ukvm-bin: exit_reason = 0x8
root@ubuntu:~/solo5/tests/test_hello# 
```

Invalid image with ELF validation:
```
root@ubuntu:~/solo5/tests/test_hello# ./ukvm-bin /tmp/osv-images/ukvm.img 
ukvm-bin: /tmp/osv-images/ukvm.img is not in ELF format
: Success
root@ubuntu:~/solo5/tests/test_hello# file /tmp/osv-images/ukvm.img 
/tmp/osv-images/ukvm.img: QEMU QCOW Image (v3), 10737418240 bytes
```

Valid image:
```
root@ubuntu:~/solo5/tests/test_hello# ./ukvm-bin test_hello.ukvm 
Setting the CPUID.
            |      ___|
  __|  _ \  |  _ \ __ \
\__ \ (   | | (   |  ) |
____/\___/ _|\___/____/
mem_size=20000000, kernel_end=10e000
Initializing the KVM Paravirtualized clock.
Hello, World
Command line is: 
solo5_app_main() returned with 0
Kernel done.
Goodbye!
KVM_EXIT_HLT
root@ubuntu:~/solo5/tests/test_hello# file test_hello.ukvm 
test_hello.ukvm: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, not stripped
```

mato commented:
Hi @myechuri. Thanks for this. Would you mind re-working it to inline the test directly into `load_code()`? Also, it might be worth adding a check for `e_ident[EI_CLASS] == ELFCLASS64 && e_type == ET_EXEC && e_machine == EM_X86_64`.

myechuri commented:
Thanks for the quick review, @mato !

> Would you mind re-working it to inline the test directly into load_code()? 

Sure, happy to. The reason i pulled it out to a function was that i thought it might be cleaner to add additional validation checks in the function in future. Not a strong reason, will rework.

> e_ident[EI_CLASS] == ELFCLASS64 e_type == ET_EXEC && e_machine == EM_X86_64

Done. Thanks for suggesting!
_______
Testing done after addressing review comments:

With invalid image:
```
root@ubuntu:~/solo5/tests/test_hello# file /tmp/osv-images/ukvm.img
/tmp/osv-images/ukvm.img: QEMU QCOW Image (v3), 10737418240 bytes
root@ubuntu:~/solo5/tests/test_hello# ./ukvm-bin /tmp/osv-images/ukvm.img
ukvm-bin: /tmp/osv-images/ukvm.img is in invalid ELF64 format
: Success
```

With valid image:
```
root@ubuntu:~/solo5/tests/test_hello# ./ukvm-bin ./test_hello.ukvm
Setting the CPUID.
            |      ___|
  __|  _ \  |  _ \ __ \
\__ \ (   | | (   |  ) |
____/\___/ _|\___/____/
mem_size=20000000, kernel_end=10e000
Initializing the KVM Paravirtualized clock.
Hello, World
Command line is: 
solo5_app_main() returned with 0
Kernel done.
Goodbye!
KVM_EXIT_HLT
root@ubuntu:~/solo5/tests/test_hello#
```

mato commented:
Hi @myechuri. A couple more nits:
- You should call `errx()` instead of `err()`. The latter appends an error message based on `errno`, which is producing the confusing `: Success` in your output.
- The error message does not need a `\n` in it, this is appended automatically. (See `man err`.)

myechuri commented:
Thanks for taking a second look, @mato.

> You should call errx() instead of err()

Thanks for catching! Fixed.

> The error message does not need a \n in it, this is appended automatically. 

Thanks for pointing out. Fixed.

mato commented:
Thanks @myechuri! Merging.

myechuri commented:
Thanks for shepherding my first solo5 PR, @mato !
